jolie/src/jolie/runtime/typing/TypePair.java

```diff
@@ -0,0 +1,84 @@
+package jolie.runtime.typing;
+
+import jolie.lang.NativeType;
+import jolie.runtime.Value;
+import jolie.util.Range;
+
+import java.util.Map;
+
+/**
+ * Created by User on 01.08.2015.
+ */
+public class TypePair extends Type {
```

Comment:
TypePair is a bit unintuitive. Either TypeChoice or TypeSum would be more standard.